src/Raven.Server/Documents/ETL/Providers/Raven/RavenEtl.cs

```diff
@@ -36,14 +40,43 @@ protected override IEnumerator<RavenEtlItem> ConvertTombstonesEnumerator(IEnumer
 
         protected override EtlTransformer<RavenEtlItem, ICommandData> GetTransformer(DocumentsOperationContext context)
         {
-            return new RavenEtlDocumentTransformer(Database, context, EtlConfiguration);
+            return new RavenEtlDocumentTransformer(Database, context, _script);
         }
 
-        protected override void LoadInternal(IEnumerable<ICommandData> commands, JsonOperationContext context)
+        protected override void LoadInternal(IEnumerable<ICommandData> items, JsonOperationContext context)
         {
-            var batchCommand = new BatchCommand(new DocumentConventions(), context, commands as List<ICommandData>);
+            var commands = items as List<ICommandData>;
+
+            Debug.Assert(commands != null);
+
+            if (commands.Count == 0)
+                return;
+
+            BatchOptions options = null;
+            if (EtlConfiguration.LoadRequestTimeoutInSec != null)
+            {
+                options = new BatchOptions
+                {
+                    RequestTimeout = TimeSpan.FromSeconds(EtlConfiguration.LoadRequestTimeoutInSec.Value)
+                };
+            }
+
+            var batchCommand = new BatchCommand(new DocumentConventions(), context, commands, options);
+            
+            try
+            {
+                _requestExecutor.Execute(batchCommand, context, CancellationToken);
+            }
+            catch (OperationCanceledException e)
```

Comment:
I just wanted to wrap `OperationCancelledException` here to ensure that we won't think it was cancellation of a ETL process and stop the processing.